feat: Add Two-Factor Authentication (2FA) TOTP Support 👀

PHPxCODER opened the pull request:
## Summary
Implements TOTP-based 2FA with authenticator apps (Google Authenticator, Authy, etc.) and backup codes for enhanced account security.

## Key Features
- **Smart Auth Flow**: Auto-detects 2FA status, bypasses email OTP when enabled
- **Setup Wizard**: QR code generation → verification → backup codes
- **Profile Integration**: Toggle 2FA on/off in user settings
- **Security**: Proper session handling, secrets never exposed to client

## Technical Changes
- Added `twoFactorEnabled`, `twoFactorSecret`, `backupCodes` to User model
- New API routes: `/api/2fa/*` and `/api/callback/2fa`
- Enhanced AuthFlow with TOTP verification components
- Updated NextAuth session handling for 2FA logins

## Dependencies
- `speakeasy` - TOTP generation/verification
- `qrcode` - QR code generation

**Result**: Optional 2FA that doesn't disrupt existing users, enterprise-grade security for those who opt-in.

PHPxCODER committed:
2FA TOTP BC 👀🤷‍♂️🙄💥

dinokage commented on package-lock.json:
```diff
@@ -5050,6 +5055,85 @@
         }
       }
     },
+    "node_modules/@radix-ui/react-separator": {
```
nice

dinokage commented on src/app/(landing)/(user)/profile/ProfileClientWrapper.tsx:
```diff
@@ -0,0 +1,214 @@
+    });
+    
+    // Optionally refresh the page to update the UI
+    window.location.reload();
```
isko hata bokadosa

PHPxCODER replied:
[masala ko hata diya bokadosa se!](https://github.com/PHPxCODER/rdp-website/commit/bf201a4d99ccd2f41ee8e4d2bea383a11b721696)

dinokage commented on src/app/(landing)/authflow/TwoFactorStep.tsx:
```diff
@@ -0,0 +1,171 @@
+            {code.map((digit, i) => (
+              <div key={i} className="flex items-center">
+                <div className="relative">
+                  <input
```
ye input change karo thoda for user experience.backspace wala chod

dinokage reviewed (commented):
check review

dinokage reviewed (changes requested):
imp

PHPxCODER committed:
masala ko hata diya bokadosa se!

Improved avatar rendering with better sizing and object-fit, updated 2FA setup/disable handlers to suggest refetching user data instead of reloading the page, and enhanced code formatting for readability. Also updated date formatting for 'Member Since' to use 'en-IN' locale.

dinokage reviewed (commented):
lgtm

dinokage commented:
hanji reopen mangta haai
